src/ceph_osd.cc

```diff
@@ -477,7 +480,7 @@ int main(int argc, const char **argv)
   if (!ms_public || !ms_cluster || !ms_hb_front_client || !ms_hb_back_client || !ms_hb_back_server || !ms_hb_front_server || !ms_objecter)
     exit(1);
   ms_cluster->set_cluster_protocol(CEPH_OSD_PROTOCOL);
-  ms_hb_front_client->set_cluster_protocol(CEPH_OSD_PROTOCOL);
```

Comment:
Nope.. we can't set the protocol on this msgr because it connects to the osd front-side interface which has no cluster protocol set (it's 0).  This is why the legacy one is introduced.